syntax = "proto3";

package tbox.proto;

message BaseConfig {
  string server_addr = 1;
  uint32 grpc_server_port = 2;
  uint32 http_server_port = 3;
  uint32 metric_ratio = 4;
  uint32 metric_interval_sec = 5;
  uint32 discard_ratio = 6;
  uint32 grpc_threads = 7;
  uint32 event_threads = 8;
  uint32 client_worker_thread_pool_size = 9;
  uint32 check_interval_seconds = 10;

  // Authentication credentials
  string user = 11;
  string password = 12;

  // Route53 settings for DNS update
  string route53_hosted_zone_id = 13;

  // Client identification
  string client_id = 14;

  // Monitor domains list for DDNS management
  repeated string monitor_domains = 15;

  // AWS credentials for Route53 access
  string aws_access_key_id = 16;
  string aws_secret_access_key = 17;
  string aws_region = 18;
}
